systemtest/src/test/java/io/strimzi/systemtest/upgrade/AbstractUpgradeST.java

```diff
@@ -91,49 +92,50 @@ public class AbstractUpgradeST extends AbstractST {
 
     protected File kafkaYaml;
 
-    protected static JsonArray readUpgradeJson(String fileName) {
+    public static List<VersionModificationData> getVersionModificationData(String filename) {
         try {
-            String jsonStr = IOUtils.toString(new FileReader(TestUtils.USER_PATH + "/src/test/resources/upgrade/" + fileName));
+            ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+            CollectionType modificationDataListType = mapper.getTypeFactory().constructCollectionType(List.class, VersionModificationData.class);
+            List<VersionModificationData> versionModificationDataList = mapper.readValue(new File(TestUtils.USER_PATH + "/src/test/resources/upgrade/" + filename), modificationDataListType);
 
-            return new JsonArray(jsonStr);
-        } catch (IOException e) {
-            e.printStackTrace();
-            throw new RuntimeException(TestUtils.USER_PATH + "/src/test/resources/upgrade/" + fileName + " file was not found.");
+            return versionModificationDataList;
+        } catch (Exception ex) {
+            LOGGER.error("Error while parsing ST data from YAML ", ex);
         }
+        return null;
     }
 
-    protected static Stream<Arguments> loadJsonUpgradeData() {
-        JsonArray upgradeData = readUpgradeJson(UPGRADE_JSON_FILE);
+    protected static Stream<Arguments> loadYamlUpgradeData() {
+        List<VersionModificationData> upgradeDataList = getVersionModificationData(UPGRADE_YAML_FILE);
         List<Arguments> parameters = new LinkedList<>();
 
         List<TestKafkaVersion> testKafkaVersions = TestKafkaVersion.getSupportedKafkaVersions();
         TestKafkaVersion testKafkaVersion = testKafkaVersions.get(testKafkaVersions.size() - 1);
 
-        upgradeData.forEach(jsonData -> {
-            JsonObject data = (JsonObject) jsonData;
+        // Generate procedures for upgrade
+        Map<String, String> procedures = new HashMap<>() {{
+                put("kafkaVersion", testKafkaVersion.version());
+                put("logMessageVersion", testKafkaVersion.messageVersion());
+                put("interBrokerProtocolVersion", testKafkaVersion.protocolVersion());
+            }};
 
-            data.put("urlTo", "HEAD");
-            data.put("toVersion", "HEAD");
-            data.put("toExamples", "HEAD");
-
-            // Generate procedures for upgrade
-            JsonObject procedures = new JsonObject();
-            procedures.put("kafkaVersion", testKafkaVersion.version());
-            procedures.put("logMessageVersion", testKafkaVersion.messageVersion());
-            procedures.put("interBrokerProtocolVersion", testKafkaVersion.protocolVersion());
-            data.put("proceduresAfterOperatorUpgrade", procedures);
+        upgradeDataList.forEach(upgradeData -> {
+            // Set latest upgrade parameters
+            upgradeData.setUrlTo("HEAD");
+            upgradeData.setToVersion("HEAD");
+            upgradeData.setToExamples("HEAD");
```

Comment:
Out of curiosity, can you set this during the `VersionModificationData` initialization?

Reply:
Will try

Reply:
so i cannot do that during init, since the class is used by up and downstream, i can do post set like this one.

Reply:
moved to Datalist class